From 55645df94f42be8991319ae78d760d3f8bbc48ed Mon Sep 17 00:00:00 2001
From: Tom Rondeau <tom@trondeau.com>
Date: Sun, 1 Dec 2013 11:46:23 -0500
Subject: [PATCH 04/25] math: fixes fast_atan2 float==0 check.

runtime: adds math functions to swig to make available as gr.function.
---
 gnuradio-runtime/lib/math/fast_atan2f.cc |    2 +-
 gnuradio-runtime/swig/runtime_swig.i     |    3 +++
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/gnuradio-runtime/lib/math/fast_atan2f.cc b/gnuradio-runtime/lib/math/fast_atan2f.cc
index 3c4967d..2cafeec 100644
--- a/gnuradio-runtime/lib/math/fast_atan2f.cc
+++ b/gnuradio-runtime/lib/math/fast_atan2f.cc
@@ -132,7 +132,7 @@ namespace gr {
     y_abs = fabsf(y);
     x_abs = fabsf(x);
     /* don't divide by zero! */
-    if((y_abs < 1.5E-5) && (x_abs < 1.5E-5))
+    if(!((y_abs > 0.0f) || (x_abs > 0.0f)))
       return 0.0;
 
     //z = (y_abs < x_abs ? y_abs / x_abs : x_abs / y_abs);
diff --git a/gnuradio-runtime/swig/runtime_swig.i b/gnuradio-runtime/swig/runtime_swig.i
index 2727f64..d4b55f1 100644
--- a/gnuradio-runtime/swig/runtime_swig.i
+++ b/gnuradio-runtime/swig/runtime_swig.i
@@ -59,6 +59,7 @@
 #include <gnuradio/tagged_stream_block.h>
 #include <gnuradio/top_block.h>
 #include <gnuradio/logger.h>
+#include <gnuradio/math.h>
 %}
 
 %constant int sizeof_char 	= sizeof(char);
@@ -92,3 +93,5 @@
 %include "gr_logger.i"
 %include "gr_swig_block_magic.i"
 %include "gr_ctrlport.i"
+
+%include "gnuradio/math.h"
-- 
1.7.10.4

